### DISCUSSION TITLE
Keycloak Usermodel id as SAML NameID

### DISCUSSION THREAD

**Discussion by sirishkumar** [keycloak/keycloak#14112] [Q&A] [answered]
We have following setup in our organisation

- Keycloak which acts as identity broker to our company IdP solution using OpenId. 
- All the services our teams owns uses SAML to authenticate users with Keycloak. 
- We only allow users we created in Keycloak to access our services. 

But in our organizations users identities constantly change, at which point we also change their username in Keycloak.  Currently we are using Username field as SAML NameID and we update the NameID associated with each service. To retain user identities on different services we are planning to use ID field in UserModel as SAML NameId. 

I want to get the community feedback on

- Whether using UserModel ID attribute as NameId is the best solution for our use case 
- Whether ID in UserModel is guaranteed to be same for a user.

**Comment by darius-m** [comment, marked as answer]
This sounds similar to #14136. Take a look if that addresses your use case.

**Reply by sirishkumar** [reply]
@darius-m  At present i can already use the User Attribute mapper for NameID as shown below. My question was whether using ID from UserModel a better practice and whether it is guaranteed to remain same.

<img width="751" alt="image" src="https://user-images.githubusercontent.com/27522/189690714-adc1746c-8d6e-41a9-b7fe-335bc871189c.png">
